### ISSUE TITLE
Crashes when requesting non-existent message-ID via API

### ISSUE THREAD

**Issue by playdohface** [mailtutan/mailtutan#75]
To reproduce, start Mailtutan locally (I did so via docker), then run 
```sh
curl localhost:1080/api/messages/1/html
``` 
and see that the server crashes:
```
thread 'tokio-runtime-worker' panicked at 'called `Option::unwrap()` on a `None` value', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/mailtutan-lib-0.3.0/src/storage/memory.rs:49:33
```
All subsequent requests will fail until the server is restarted:
```
thread 'tokio-runtime-worker' panicked at 'called `Result::unwrap()` on an `Err` value: PoisonError { .. }', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/mailtutan-lib-0.3.0/src/api/messages.rs:10:36
```
I think that the correct behavior here would be an HTTP 404 response and the server continuing to run as normal. 

I am working on a fix already. (`Storage.get(id)` should return `Option<Message>` instead of `Message` and absence of values should be handled)

**Comment by stappersg** [comment]
On Sat, Aug 10, 2024 at 02:33:29AM -0700, playdohface wrote:
> I am working on a fix already.

Acknowledge


Groeten
Geert Stappers
In an atttempt to express "the reported issue got human attention".
-- 
Silence is hard to parse

**Comment by stappersg** [comment]
Over here on Debian GNU/Linux self build from main, I get different behaviour as reported.

Main difference is getting each time same log output.

```text
stappers@paddy:~/src/githubforks/mailtutan
$ listening on smtp://0.0.0.0:1025
listening on http://0.0.0.0:1080

stappers@paddy:~/src/githubforks/mailtutan
$ curl localhost:1080/api/messages/1/html
thread 'tokio-runtime-worker' panicked at mailtutan-lib/src/storage/memory.rs:49:33:
called `Option::unwrap()` on a `None` value
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
curl: (52) Empty reply from server
stappers@paddy:~/src/githubforks/mailtutan
$ curl localhost:1080/api/messages/1/html
thread 'tokio-runtime-worker' panicked at mailtutan-lib/src/storage/memory.rs:49:33:
called `Option::unwrap()` on a `None` value
curl: (52) Empty reply from server
stappers@paddy:~/src/githubforks/mailtutan
$ curl localhost:1080/api/messages/1/html
thread 'tokio-runtime-worker' panicked at mailtutan-lib/src/storage/memory.rs:49:33:
called `Option::unwrap()` on a `None` value
curl: (52) Empty reply from server
stappers@paddy:~/src/githubforks/mailtutan
$ git branch | grep ^\*
* main
stappers@paddy:~/src/githubforks/mailtutan
$ 
```

**Comment by stappersg** [comment]
Reporting another observation:
```text
stappers@paddy:~/src/githubforks/mailtutan
$ listening on smtp://0.0.0.0:1025
listening on http://0.0.0.0:1080

stappers@paddy:~/src/githubforks/mailtutan
$ sudo ss -tlp | grep -e 1025 -e 1080
LISTEN 0      128                    0.0.0.0:1025        0.0.0.0:*    users:(("mailtutan",pid=10044,fd=14))                                                                                                                              
stappers@paddy:~/src/githubforks/mailtutan
$ 
```
Note that `ss` does not see something listing on port 1080.
